### ISSUE TITLE
Weird issues with wallpaper

### ISSUE THREAD

**Issue by anowlcalledjosh** [yshui/compton#107]
### Platform

Ubuntu Desktop 18.04 amd64

### GPU, drivers, and screen setup

Nvidia GTX 1050 (driver version 410.78) & Intel integrated graphics, Mesa 18.2, two monitors (external above, internal below)

<details><summary><code>glxinfo -B</code> (using Intel graphics)</summary>

```
$ glxinfo -B
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Intel Open Source Technology Center (0x8086)
    Device: Mesa DRI Intel(R) UHD Graphics 630 (Coffeelake 3x8 GT2)  (0x3e9b)
    Version: 18.2.2
    Accelerated: yes
    Video memory: 3072MB
    Unified memory: yes
    Preferred profile: core (0x1)
    Max core profile version: 4.5
    Max compat profile version: 3.0
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
OpenGL vendor string: Intel Open Source Technology Center
OpenGL renderer string: Mesa DRI Intel(R) UHD Graphics 630 (Coffeelake 3x8 GT2) 
OpenGL core profile version string: 4.5 (Core Profile) Mesa 18.2.2
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 3.0 Mesa 18.2.2
OpenGL shading language version string: 1.30
OpenGL context flags: (none)

OpenGL ES profile version string: OpenGL ES 3.2 Mesa 18.2.2
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```

</details>

### Environment

i3-gaps 4.16.1

### Compton version

v5-rc4-71-g5fffb6f3f

```
$ src/compton-next/build/src/compton --diagnostics
[ 08/02/19 12:55:29.387 parse_config_libconfig WARN ] "paint-on-overlay" has been removed as an option, and is enabled whenever possible
**Version:** v5-rc4-71-g5fffb6f3f

### Extensions:

* Name Pixmap: Yes
* Shape: Yes
* XRandR: Yes
* Present: Present

### Misc:

* Use Overlay: Yes
* Config file used: /home/josh/.config/compton.conf
```

### Compton configuration:

```
# Can be "xrender" or "glx"; GLX is faster (usually) but needs a sane driver.
backend = "glx";

# Magically improve performance. Can allegedly cause issues with transparency.
glx-no-stencil = true;

# Possible big performance boost on partial screen updates, but breaks with some drivers.
glx-no-rebind-pixmap = true;

# Copy stuff instead of redrawing it when unmodified. Can cause slowdown on some systems.
# glx-copy-from-front = true;

# Set this to 0 for maximum sanity (and slowness), 1 for fast but maybe broken, and 2-6 for increasing stability. 3 is approximately triple-buffered.
# "buffer-age" (or -1) apparently auto-detects it.
# 1 is broken with the modesetting driver on Intel graphics.
# 2 is also broken in that configuration (but a little less so).
glx-swap-method = 3;

# Can be "none", "drm", "opengl", "opengl-oml", "opengl-swc", or "opengl-mswc".
vsync = "opengl-swc";

# Attempt to unredirect when a fullscreen window is detected for improved performance.
unredir-if-possible = true;
# Maybe improve flickering on unredirect???
paint-on-overlay = true;
```

### Steps of reproduction

1. Restart the computer
2. Attach the external monitor
3. Start compton

### Expected behavior

The wallpaper shouldn't change when I start compton

### Current Behavior

The wallpaper on the internal monitor changes to the bottom row of pixels of the wallpaper, stretched vertically across the whole screen.

If I set the wallpaper again using `feh` (with the exact same command as is run on every boot, `feh --no-fehgb --bg-fill image.png`), then this bug stops happening until I reboot again.

Using the xrender backend (`compton --config /dev/null --backend xrender`) works fine – on the original compton 0.1~beta2+20150922, it does the Windows XP thing where dragging a window around won't redraw behind the window so you get a trail across the screen, but that doesn't seem to happen on version v5-rc4-71-g5fffb6f3f.

This bug previously filed as chjj/compton#538.

### Stack trace

n/a

### Other details

Example screenshot:

![image](https://user-images.githubusercontent.com/9433472/52449988-1ce9ff80-2b31-11e9-8131-31dcfd8c7a40.png)

**Comment by yshui** [comment]
Can you check if the `wallpaper-fix` branch fix your problem?

**Comment by anowlcalledjosh** [comment]
Yep, that fixes it. Thanks!

**Comment by Bracket-I-I** [comment]
Is the 'wallpaper fix" branch change in 12.5? Cause I still have this issue.
`picom --backend glx --vsync `
I have to rerun feh to undo the smear.

**Comment by kirillin** [comment]
The same problem in picom 12.5, ubuntu 24.04 (and 25.04) with i3wm on lenovo thinkpad yoga gen 4 with intel graphics only. 

I did try change flags and backends of picom config, but it did not solve the issue. However, if I run feh in terminal, it repairs. 

My `~/.config/picom $ cat picom.conf`
```
backend = "glx";
glx-no-stencil = true;
glx-no-rebind-pixmap = true;
vsync = true;

inactive-opacity = 0.9;
active-opacity = 0.9;
frame-opacity = 0.9;
blur-background = true;

opacity-rule = [
    "100:class_g = 'Evince'",
    "100:class_g = 'firefox_firefox'",
    "100:class_g = 'Code'"
];
```
